File: LeetCode/2027. Minimum Moves to Convert String/solution.py
"""
Runtime: 40 ms
Memory Usage: 14.9 MB
"""
class Solution:
    def minimumMoves(self, s: str) -> int:
        move = 0
        cover = -1
        for i, c in enumerate(s):
            if c == 'X' and i > cover:
                move += 1
                cover = i + 2
        return move
        

File: LeetCode/2027. Minimum Moves to Convert String/solution.cpp
/**
 * Runtime: 0 ms
 * Memory Usage: 6.3 MB 
 */
class Solution {
public:
    int minimumMoves(string s) {
        int move = 0, cover = -1;
        for (int i = 0; i < s.size(); ++ i) {
            if (s[i] == 'X' && i > cover) {
                ++ move;
                cover = i + 2;
            }
        }
        return move;
    }
};
